ERL-991 Strip control codes from eunit_surefire output

ampleyfly opened the pull request:
Without this, test cases which output control codes, or with names or
descriptions containing control codes, would cause the generated XML files to be
invalid.

Backport from master.

IngelaAndin commented:
Please rebase your branch on maint and we will include it in the last planned patch release OTP-22.3 
The maint-22 branch is only used for unplanned patches, that are mainly done for serious bugs, vulnerabilities or features considered important by our product manager.

claassistantio commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/erlang/otp?pullRequest=2487) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/erlang/otp?pullRequest=2487) before we can accept your contribution.<br/>**10** out of **12** committers have signed the CLA.<br/><br/>:white_check_mark: essen<br/>:white_check_mark: sverker<br/>:white_check_mark: IngelaAndin<br/>:white_check_mark: sirihansen<br/>:white_check_mark: attah<br/>:white_check_mark: kjellwinblad<br/>:white_check_mark: bjorng<br/>:white_check_mark: peterdmv<br/>:white_check_mark: HansN<br/>:white_check_mark: ampleyfly<br/>:x: proxyles<br/>:x: bmk<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/erlang/otp?pullRequest=2487) it.</sub>

ampleyfly commented:
So, I rebased my branch on maint and changed the target branch to maint.
I don't know what's up with the CLA check, I have signed the CLA and the check passed the last time.

IngelaAndin commented:
Thanks, I  have put it in testing. I think that the CLA check might have been failing because of the branch being based on something else then maint, it can make it look like other commits than yours where part of the PR.

IngelaAndin commented:
Thank you for your contribution. It has passed the test phase and is now merged to maint.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/erlang/otp?pullRequest=2487) <br/>All committers have signed the CLA.
